src/Metadata/Property/Factory/AnnotationSubresourceMetadataFactory.php

```diff
@@ -82,7 +82,8 @@ private function updateMetadata(ApiSubresource $annotation, PropertyMetadata $pr
         $type = $propertyMetadata->getType();
         $isCollection = $type->isCollection();
         $resourceClass = $isCollection ? $type->getCollectionValueType()->getClassName() : $type->getClassName();
+        $maxDepth = $annotation->maxDepth ? $annotation->maxDepth : null;
```

Comment:
Do we really want to turn 0 into null?

Reply:
depth 0 has no sens I think. If it's 0 or null, we do not apply maxDepth. If maxDepth=1 we create one subresource operation, if maxDepth=2 we create two subresources depth, ...

Reply:
null is correct here yes

Reply:
Right, the behaviour should be the same but IMO we should not transform the configured value: 0 is not null (it's important in terms of extension point). What do you think about keeping 0? The behavior will remain the same with your condition in `SubresourceOperationFactory::computeSubresourceOperations()`.

Reply:
Good point! fixed